app/controllers/amusement_parks_controller.rb

```diff
@@ -0,0 +1,11 @@
+class AmusementParksController < ApplicationController 
+    def show 
+        @amusement_park = AmusementPark.find(params[:id])
+        @mechanic_list = []
+        @amusement_park.rides.each do |ride|
```

Comment:
We'd rather you use AR here than Ruby to process this data! 

In the AmusementPark model, you already have `has_many :rides` so you can leverage this association and add `has_many :mechanics, through: :rides` which will allow you do add a model method to grab all the mechanics for each amusement park. The model method can contain `self.mechanics.distinct` where self is the instance of the amusement park